### ISSUE TITLE
Docs: Clarify application.yaml source support with MP Config

### ISSUE THREAD

**Issue by arjav-desai** [helidon-io/helidon#7482]
A user has reported that
"The documentation of Helidon MP Config mentions System properties, Environment variables, and /META-INF/microprofile-config.properties (https://helidon.io/docs/v3/#/mp/config/introduction). I have noticed though that an application.yaml in the classpath is also loaded. Is this behavior expected or is it something in my project setting this up? thank you"

We have confirmed that
"Helidon MP configuration, when using the default (e..g you just call ConfigProvider.getConfig() ), we create an instance based on the MP specification - we look for implemented sources using ServiceLoader.
It also looks for a Helidon specific config source provider, as that has support for “meta configuration”.
Our io.helidon.config.yaml.mp.YamlConfigSourceProvider provides this services, and as long as its module is on the classpath, it will be discovered. And this provider adds application.yaml as a source"

So basically,
We need to clarify that if user has application.yaml on their classpath, it will be loaded and used by default. If there are config properties/values which exists in multiple config source (say application.yaml and microprofile-config.properties) than as of now the order of precedence is based on classloader. However, we are filing another issue to look into having a deterministic order.
